static/js/containers/pages/DashboardPage_test.js

```diff
@@ -82,6 +85,21 @@ describe("DashboardPage", () => {
     assert.isFalse(unlinkedTitle.find("a").exists())
   })
 
+  it("Not links to the courseware URL if run is started in future", async () => {
```

Comment:
This test case and the test case above can be replaced with a single test if you stub the `isLinkableCourseRun` function. `static/js/containers/App_test.js` has an example where we stub an API method if you want an example. It would look something like this

```es6
import * as courseApi from "../../lib/courseApi"
//...
describe("DashboardPage", () => {
  let isLinkableStub
  beforeEach(() => {
    isLinkableStub = helper.sandbox.stub(courseApi, "isLinkableCourseRun")
    //...
  })

  //
  ;[
    [false, false],
    [true, true]
  ].forEach(([isLinkable, expCourseLink]) => {
    it(`${shouldIf(expCourseLink)} show a link if course run ${isIf(isLinkable) linkable}`, async () => {
      isLinkableStub.returns(isLinkable)
      const future = moment().add(10, "days")
      const exampleCoursewareUrl = "http://example.com/my-course"
      const enrollment = {
        ...userEnrollments[0],
        courseware_url: exampleCoursewareUrl
      }
      const { inner } = await renderPage({
        entities: { enrollments: [enrollment] }
      })
      const enrolledItems = inner.find(".enrolled-item")
      assert.lengthOf(enrolledItems, 1)
      assert.equal(enrolledItems.at(0).find("a").exists(), expCourseLink)
      if (expCourseLink) {
        assert.equal(enrolledItems.at(0).prop("href"), exampleCoursewareUrl)
      }
    })
  })
})

```

Parametrizing the test uses less code and it illustrates our expectations about these links a little more clearly

Reply:
That is good suggestion for writing JS tests in more well way. I like it but earlier I was trying to set with the enrollment dates while rendering the page to test behavior instead of Stubbing the `isLinkableCourseRun ` True/False.

Reply:
Did some modifications in the actual suggestions. 

```
;[[false, false], [true, true]].forEach(([isLinkable, expCourseLink]) => {
    it(`${shouldIf(expCourseLink)} show a link if course run ${isIf(
      isLinkable
    )} linkable`, async () => {
      isLinkableStub.returns(isLinkable)
      const exampleCoursewareUrl = "http://example.com/my-course"
      userEnrollments[0].run.courseware_url = exampleCoursewareUrl
      const enrollment = {
        ...userEnrollments[0]
      }
      const { inner } = await renderPage({
        entities: { enrollments: [enrollment] }
      })
      const enrolledItems = inner.find(".enrolled-item")
      assert.lengthOf(enrolledItems, 1)
      assert.equal(
        enrolledItems
          .at(0)
          .find("a")
          .exists(),
        expCourseLink
      )
      if (expCourseLink) {
        const linkedTitle = enrolledItems.at(0).find("h4")
        assert.equal(linkedTitle.find("a").prop("href"), exampleCoursewareUrl)
      }
    })
  })
```

Reply:
my bad, I forgot to remove it.